includes/class-admin-settings.php

```diff
@@ -0,0 +1,516 @@
+	}
+
+	/**
+	 * Register the Admin Menu.
+	 *
+	 * @return void
+	 */
+	public function register_admin_menu() {
+		add_options_page(
+			'AspirePress',
+			'AspirePress',
+			'manage_options',
+			'aspirepress-settings',
+			array( $this, 'the_settings_page' )
+		);
+	}
+
+	/**
+	 * Enqueue the Scripts and Styles.
+	 *
+	 * @param string $hook The page identifier.
+	 * @return void
+	 */
+	public function admin_enqueue_scripts( $hook ) {
+		if ( 'settings_page_aspirepress-settings' !== $hook ) {
+			return;
+		}
+		wp_enqueue_style( 'aspire_update_settings_css', plugin_dir_url( __DIR__ ) . 'assets/css/aspire-update.css', array(), '1.0' );
+		wp_enqueue_script( 'aspire_update_settings_js', plugin_dir_url( __DIR__ ) . 'assets/js/aspire-update.js', array( 'jquery' ), '1.0', true );
+		wp_localize_script(
+			'aspire_update_settings_js',
+			'aspirepress',
+			array(
+				'ajax_url'       => admin_url( 'admin-ajax.php' ),
+				'nonce'          => wp_create_nonce( 'aspirepress-ajax' ),
+				'domain'         => Utilities::get_top_level_domain(),
+				'apikey_api_url' => 'api.aspirepress.org/repository/api/v1/apitoken',
+			)
+		);
+	}
+
+	/**
+	 * The Settings Page Markup.
+	 *
+	 * @return void
+	 */
+	public function the_settings_page() {
+		$reset_url = add_query_arg(
+			array(
+				'reset'       => 'reset',
+				'reset-nonce' => wp_create_nonce( 'aspirepress-reset-nonce' ),
+			),
+			admin_url( 'options-general.php?page=aspirepress-settings' )
+		);
+		?>
+		<div class="wrap">
+			<h1><?php esc_html_e( 'AspirePress Settings', 'aspirepress' ); ?></h1>
+			<form method="post" action="options.php">
+				<?php
+				settings_fields( $this->option_group );
+				do_settings_sections( 'aspirepress-settings' );
+				?>
+				<p class="submit">
+					<input type="submit" name="submit" id="submit" class="button button-primary" value="Save Changes">
+					<a href="<?php echo esc_url( $reset_url ); ?>" class="button button-secondary" >Reset</a>
+				</p>
+			</form>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Register all Settings.
+	 *
+	 * @return void
+	 */
+	public function register_settings() {
+		$nonce   = wp_create_nonce( 'aspirepress-settings' );
+		$options = get_option( $this->option_name );
+
+		register_setting(
+			$this->option_group,
+			$this->option_name,
+			array(
+				'sanitize_callback' => array( $this, 'sanitize_settings' ),
+			)
+		);
+
+		add_settings_section(
+			'aspirepress_settings_section',
+			__( 'API Configuration', 'aspirepress' ),
+			null,
+			'aspirepress-settings',
+			array(
+				'before_section' => '<div class="%s">',
+				'after_section'  => '</div>',
+			)
+		);
+
+		add_settings_field(
+			'enable',
+			'Enable AspirePress API Rewrites',
+			array( $this, 'add_settings_field_callback' ),
+			'aspirepress-settings',
+			'aspirepress_settings_section',
+			array(
+				'id'   => 'enable',
+				'type' => 'checkbox',
+				'data' => $options,
+			)
+		);
+
+		add_settings_field(
+			'api_key',
+			__( 'API Key', 'aspirepress' ),
```

Comment:
Just remember that every thing that is sent for translation can come back with dirty data. Best use esc_html__() over __() whenever applicable or possible or in doubt. Translations are generally one of those places.

Reply:
updated to use esc_html__